From 7bf7222d4dc6b0a8a23b4a3be52658a8389ff9ff Mon Sep 17 00:00:00 2001
From: Aaron Bassett <abassett@qnx.com>
Date: Mon, 2 Mar 2026 16:15:41 +0000
Subject: [PATCH] qnx suppport

Imported from qnx-ports/avahi (branch qnx-0.9.x, db5a8e5..7bf7222); original authorship kept.
Upstream-Status: QNX-specific (carried in the qnx-ports fork)

---
 avahi-common/Makefile.am                  | 4 ++++
 avahi-compat-howl/include/salt/platform.h | 6 +++++-
 avahi-daemon/main.c                       | 4 ++++
 avahi-utils/sigint.c                      | 4 ++++
 4 files changed, 17 insertions(+), 1 deletion(-)

diff --git a/avahi-common/Makefile.am b/avahi-common/Makefile.am
index bba2883..60faf76 100644
--- a/avahi-common/Makefile.am
+++ b/avahi-common/Makefile.am
@@ -20,6 +20,10 @@ AM_CFLAGS=-I$(top_srcdir)
 # This cool debug trap works on i386/gcc only
 AM_CFLAGS+='-DDEBUG_TRAP=__asm__("int $$3")'
 
+# Qnx malloc definitions exist in malloc.h avahi-common/malloc.h is overriding malloc.h
+# we NEED this here to remove the -I. which is causing the bug
+AUTOMAKE_OPTIONS = nostdinc
+
 avahi_commonincludedir=$(includedir)/avahi-common
 
 avahi_commoninclude_HEADERS = \
diff --git a/avahi-compat-howl/include/salt/platform.h b/avahi-compat-howl/include/salt/platform.h
index 6bb9dcc..5704142 100644
--- a/avahi-compat-howl/include/salt/platform.h
+++ b/avahi-compat-howl/include/salt/platform.h
@@ -77,7 +77,11 @@ typedef _int64						u_int64_t;
 
 #endif
 
-#if defined(__sun)
+#if defined(__QNX__)
+#	include <stdint.h>
+#endif
+
+#if defined(__sun) || defined(__QNX__)
 
 #	define u_int8_t	uint8_t
 #	define u_int16_t	uint16_t
diff --git a/avahi-daemon/main.c b/avahi-daemon/main.c
index 51c28ee..b4ed0c8 100644
--- a/avahi-daemon/main.c
+++ b/avahi-daemon/main.c
@@ -90,6 +90,10 @@
 #include "dbus-protocol.h"
 #endif
 
+#ifndef SA_RESTART
+#define SA_RESTART 0
+#endif
+
 AvahiServer *avahi_server = NULL;
 AvahiSimplePoll *simple_poll_api = NULL;
 static char *argv0 = NULL;
diff --git a/avahi-utils/sigint.c b/avahi-utils/sigint.c
index b726654..ee20c1d 100644
--- a/avahi-utils/sigint.c
+++ b/avahi-utils/sigint.c
@@ -32,6 +32,10 @@
 #include <avahi-common/gccmacro.h>
 #include "sigint.h"
 
+#ifndef SA_RESTART
+#define SA_RESTART 0
+#endif
+
 static AvahiSimplePoll *simple_poll = NULL;
 static struct sigaction old_sigint_sa, old_sigterm_sa;
 static int pipe_fds[2] = { -1, -1 };
-- 
2.43.0

